mos_tests/neutron/python_tests/test_dvr.py

```diff
@@ -253,3 +259,344 @@ def test_routing_after_reset_computes(self, prepare_openstack, env_name):
+
+        Steps:
+            1. Check that the router is not distributed by default
+            2. Try to change the distributed parameter
+                It shouldn't be possible without
+                setting admin_state_up to False
+            3. Change admin_state_up to False and than set distributed to True
+            4. Change admin_state_up to True to enable the router
+            5. Check that the router namespace is available on the compute
+            6. Ping 8.8.8.8 from vm1
+            7. And finally as a bonus check that it is not poissible
+                to change the router type from distributed to centralized
+        """
+
+        # Check that router is not distributed by default
+        router = self.os_conn.neutron.list_routers(
+                    id=self.router_id)['routers'][0]
+        err_msg = 'distributed parameter for the router {0} is {1}.'.format(
+                   router['name'], router['distributed'])
+        err_msg = err_msg + " But it's expected value is False"
+        assert not router['distributed'], err_msg
+
+        # Try to change the distributed parameter
+        # That shouldn't be possible without setting admin_state_up to False
+        with pytest.raises(NeutronClientException) as e:
+            self.os_conn.neutron.update_router(self.router_id,
+                                           {'router': {
+                                            'distributed': True}})
+
+        allowed_msg = 'admin_state_up to False prior to upgrade'
+        err_msg = 'Failed to update the router, exception: {}'.format(e)
+        assert allowed_msg in str(e.value), err_msg
+
+        # Change admin_state_up to False and than set distributed to True
+        self.os_conn.neutron.update_router(self.router_id,
+                                           {'router': {
+                                            'admin_state_up': False}})
+
+        self.os_conn.neutron.update_router(self.router_id,
+                                           {'router': {
+                                            'distributed': True}})
+
+        # Change admin_state_up to True to enable the router
+        self.os_conn.neutron.update_router(self.router_id,
+                                           {'router': {
+                                            'admin_state_up': True}})
+
+        # Check that distributed is really changed to True
+        router = self.os_conn.neutron.list_routers(
+                    id=self.router_id)['routers'][0]
+        err_msg = 'distributed parameter for the router {0} is {1}.'.format(
+                   router['name'], router['distributed'])
+        err_msg = err_msg + " But it's expected value is True"
+        assert router['distributed'], err_msg
+
+        # Check that the router namespace is available on the compute now
+        with self.env.get_ssh_to_node(self.compute_ips[0]) as remote:
+            cmd = "ip netns | grep [q]router-{}".format(self.router_id)
```

Comment:
grep returns non-zero exit code if nothing find.
You can use `remote.check_call(cmd)`